### PR TITLE
Fix wiki template and font file paths

### PR DESCRIPTION
**Pull request by CyberiaResurrection**
Running PyRoute on a Fedora box, when it was originally written on a Ubuntu box, has rumbled some Ubuntu-specific assumptions as to font locations.  Running PyRoute in an IDE, to debug and profile it, rumbled an implicit assumption that the current working directory is the project root dir - that assumption didn't necessarily hold.

To fix the working dir assumption, I've tweaked the wiki template class to supply the _absolute_ base dir for the wiki templates.

To encapsulate the font-location assumptions, I've added a dirt-simple adapter class (named, with much originality, FontLayer) that translates the four font files used in SubsectorMap2.py into the underlying absolute paths on Ubuntu or Fedora.  I've structured FontLayer to make it very easy to add additional mappings for existing files (such as on other Linux distros), as well as mappings for new files.

### CODE REVIEW

**Comment by CyberiaResurrection** [comment]
@tjoneslo , this might partly solve #18 - I didn't try for a graceful fallback (given one currently doesn't exist).  Instead, I simply went for brute force over ignorance to get current master HEAD + this PR running otherwise unmodified on Fedora 33.

**Review by tjoneslo — APPROVED** [review]
If this works for you. I'll need to test it locally, and that's easier done after merged.

**Comment by CyberiaResurrection** [comment]
Thank you muchly - I tried to structure the adapter to make specific changes much easier to make, and to provide a single point of truth.

**Comment by tjoneslo** [comment]
Thank you very much for taking the time to fix these issues. It makes everyone's life much better. I am not able to push changes back to this project despite have done some work on it. It's nice to see someone continuing to keep it alive.

**Comment by CyberiaResurrection** [comment]
@tjoneslo , glad to help.  I take it local testing rumbled no major goofs on my part?

Well, fair is fair - you've previously said you don't have ready access to a Fedora box, so those blow ups were far more my problem than yours.  I was just trying to make future detonations (say a bug report from someone on FreeBSD) significantly easier to fix.

**Comment by LeamHall** [comment]
I'm working on a Fedora Dockerfile for this, but running:

  `python3 -m unittest`

Gives a lot of ModuleNotFoundErrors. How are you running the tests? I theory, if I can figure this out, we can build multiple docker files for testing.

**Comment by LeamHall** [comment]
> Thank you very much for taking the time to fix these issues. It makes everyone's life much better. I am not able to push changes back to this project despite have done some work on it. It's nice to see someone continuing to keep it alive.

Is that "not able to push changes back" a technical issue I can help with, or a time/resource issue?

**Comment by LeamHall** [comment]
https://github.com/LeamHall/docker_pyroute

**Comment by CyberiaResurrection** [comment]
@LeamHall , I've set up a venv on my Fedora box inside where I've cloned out the repo, and are running tests from the repo root dir

**Comment by LeamHall** [comment]
> @LeamHall , I've set up a venv on my Fedora box inside where I've cloned out the repo, and are running tests from the repo root dir

I apologize for not being clear. What command(s) are you using to run your tests? I tried "python3 -m unittest" in the project root directory and it could not find the modules.

**Comment by CyberiaResurrection** [comment]
@LeamHall , I'm just running **python3 -m unittest** with no problems.

**Comment by LeamHall** [comment]
@CyberiaResurrection, it may well be operator error on my part. Here's what I'm getting:

https://gist.github.com/LeamHall/5019bd8218c227096f9a542a80859c87

**Comment by CyberiaResurrection** [comment]
@LeamHall , this is a bit of a long short - can you try modifying the intra-package import statements in the test classes like so:

`from PyRoute.Star import Star`

Does that help?

**Comment by CyberiaResurrection** [comment]
I'd suggest, at a guess, chase down those breaks.  There are `sys.path.append('../PyRoute')` calls in most class files before the class definition, but I'm not sure if they're working properly

**Comment by tjoneslo** [comment]
> > Thank you very much for taking the time to fix these issues. It makes everyone's life much better. I am not able to push changes back to this project despite have done some work on it. It's nice to see someone continuing to keep it alive.
> 
> Is that "not able to push changes back" a technical issue I can help with, or a time/resource issue?

My employer does not allow contributions to open source projects without approval. Getting approval for contribution to personal projects is never approved. Not rejected either, just never approved.

**Comment by simon-hibbs** [comment]
That really sucks @tjoneslo. This projects looks interesting, I'll spin up a Linux VM and give it a go. I used NetworkX for the trade routes in StarBase. What should I go for, Fedora 35?

**Comment by CyberiaResurrection** [comment]
@simon-hibbs , I'd appreciate that (Fedora 35) to check if my font fixes work on later versions of Fedora.

Are you both familiar enough with another Linux distro (that's neither Fedora nor Ubuntu) and able to spin up a VM of that third distribution, to see where the font files are on _that_ one?  If so, could you do so and send up another PR if there's any changes needed?

**Comment by LeamHall** [comment]
@CyberiaResurrection, I can spin up a lot of things. Are you sure you want to support finding fonts on a lot of different operating systems? It can become a lot of work.

**Comment by CyberiaResurrection** [comment]
@LeamHall , I didn't know it was such a can of worms.  As such, I'm not sure I'm competent to answer - @tjoneslo would have to answer.

**Comment by tjoneslo** [comment]
Added a note to the base issue re: how to find the system fonts. We should look further to address these issues.

**Pull request merged**
